bluepyemodel/icselector/modules/gene_selector.py

```diff
@@ -0,0 +1,229 @@
+        """Get gbar_max for given gene.
+
+        Args:
+            gene (DataFrame): single column from the gene table
+
+        Returns:
+            gbar_max (float): gbar_max value
+        """
+
+        try:
+            gbar_max = gene["g_bar_max"]
+            gbar_max = np.array([float(g) for g in gbar_max.values])
+            # gbar_max = np.array(gbar_max.values, dtype = np.float)
+        except KeyError:
+            gbar_max = np.array([0.0])
+        crit = np.array([np.isnan(g) for g in gbar_max])
+        gbar_max[crit] = 0.0
+        gbar_max = np.unique(gbar_max)
+        if len(gbar_max) > 1:
+            logging.warning(
+                "Conflicting g_bar_max values for gene %s. Selected '%s' from %s",
+                gene.name,
+                np.max(gbar_max),
+                gbar_max,
+            )
+        return np.max(gbar_max)
+
+    def get(self, gene_name):
+        """Get info for given gene.
+
+        Args:
+            gene_name (str): name of gene
+
+        Returns:
+            (dict): gene info
+        """
+
+        if gene_name in self.selected_genes:
+            return self.selected_genes[gene_name]
+        raise KeyError("Gene not available for the selected ttype.")
+
+    def select_from_ttype(self, keys=None, group_compartments=False):
+        """Returns selected genes and distributions associated with provided
+        key words.
+
+        Args:
+            keys (list [str]): List of keywords
+            group_compartments: Option to combine compartments into groups
+            (e.g. 'all', or 'alldend')
+
+        Returns:
+            selected_genes (list [dict]): list of selected genes with info
+        """
+
+        df = self._gene_map
+        if isinstance(keys, str):
+            keys = [keys]
+        if keys:
+            df = self._filter(df, keys)
+        # Store result
+        self.selected_met_types = np.unique([f"{v[0]} - {v[1]}" for v in df.index.values])
+        df = df.droplevel([0, 1])
+        # Apply filter also to genes
+        if keys:
+            genes = self._filter(df.T, keys)
+            genes = genes.T
+        else:
+            genes = df
+
+        names = genes.columns.values
+        for name in names:
+            gene = genes[name]
+            # Check if gene is present
+            presence = self._get_gene_presence(gene)
+            if sum(presence) == 0:
+                continue
+            # Determine distributions per compartment
+            distros = self._get_distributions(gene, presence, group_compartments)
+            # Determine gbar_max
+            gbar_max = self._get_gbar_max(gene)
+            # Add item
+            self.selected_genes[name] = {
+                "channel": "n/a",
+                "distribution": distros,
+                "gbar_max": gbar_max,
+            }
+
+        return self.selected_genes
+
+    def __str__(self):
+        # heading_str = '\n===========\nM/E/T types\n==========='
```

Comment:
can remove those comments

Reply:
Done in [942a287](https://github.com/BlueBrain/BluePyEModel/pull/144/commits/942a287eb332cb58b5608511616eefcd2e76e880)